src/context.ts

```diff
@@ -186,43 +186,38 @@ const checker: StaticHas = {
     },
     command(command) {
         const hasEntities = checker.filterQuery(":entities:bot_command");
-        const atCommands = new Set<string>();
-        const noAtCommands = new Set<string>();
-        toArray(command).forEach((cmd) => {
+        const noAtCommands = new Set<string>(toArray(command ?? []));
+        for (const cmd of noAtCommands) {
             if (cmd.startsWith("/")) {
                 throw new Error(
                     `Do not include '/' when registering command handlers (use '${
                         cmd.substring(1)
                     }' not '${cmd}')`,
                 );
             }
-            const set = cmd.includes("@") ? atCommands : noAtCommands;
-            set.add(cmd);
-        });
+        }
         return <C extends Context>(ctx: C): ctx is CommandContext<C> => {
             if (!hasEntities(ctx)) return false;
             const msg = ctx.message ?? ctx.channelPost;
             const txt = msg.text ?? msg.caption;
-            return msg.entities.some((e) => {
-                if (e.type !== "bot_command") return false;
-                if (e.offset !== 0) return false;
-                const cmd = txt.substring(1, e.length);
-                if (noAtCommands.has(cmd) || atCommands.has(cmd)) {
-                    ctx.match = txt.substring(cmd.length + 1).trimStart();
-                    return true;
-                }
-                const index = cmd.indexOf("@");
-                if (index === -1) return false;
+            const e: MessageEntity = msg.entities[0];
```

Comment:
```suggestion
            const e : MessageEntity = msg.entities[0] ?? msg.caption_entities?.[0];
```
would also close #686

Reply:
No. The order of message entities is unspecified. The bot command at index 0 might be the last entity.

Reply:
So using `entities[0]` is error prone, for example a msg like `/start /end` could match `/end` using this approach, if I'm not mistaken

Reply:
Exactly